sklearn/cluster/_kmeans.py

```diff
@@ -897,10 +897,9 @@ def _check_params_vs_input(self, X, default_n_init=None):
             )
             self._n_init = default_n_init
         if self._n_init == "auto":
-            if self.init == "k-means++":
+            self._n_init = default_n_init
+            if (not _is_arraylike_not_scalar(self.init)) and self.init == "k-means++":
```

Comment:
Hi @glemaitre ... OK, so I added the changes you mention.. (Tho I changed it a little bit from what you wrote.. I more explicitly demand that self.init be _either_ "k-means++" _or_ "random" if it's a string, and either a callable or array like otherwise, and it raises an Exception otherwise.) I also added the docstring changes and the test case, but I don't know what I'm expected to do with the `doc/whats_new/v1.3.rst`... Can you please tell me what exactly I should be doing there?

Thanks!

Reply:
Actually, I think I kind of figured out the Changelog thing. I still don't know tho, if the _text_ in there is any good..